docs/python/programming-guide.md

```diff
@@ -7,236 +7,386 @@ pagination_prev: quickstart
-cd greeting-guestbook
+python3 -m venv dbos-starter/.venv
+cd dbos-starter
 .venv\Scripts\activate.bat
 ```
 </TabItem>
 </Tabs>
 
-Then, install and initialize DBOS:
+Then, install DBOS and create a DBOS configuration file:
 ```shell
 pip install dbos
-dbos init
+dbos init --config
 ```
 
-Now, let's use FastAPI to write a simple app that greets our friends.
-Every time the app receives a greeting, it performs two steps:
+## 2. Workflows and Steps
 
-1. Sign an online guestbook with the greeting.
-2. Record the greeting in the database.
+Now, let's create the simplest interesting DBOS program.
+Create a `main.py` file and add this code to it:
 
-We deliberately **won't** use DBOS yet (except to fetch the database connection string) so we can show you how easy it is to add later.
+```python showLineNumbers title="main.py"
+from dbos import DBOS
 
-Copy the following code into `greeting_guestbook/main.py`, replacing its existing contents:
+DBOS()
 
-```python showLineNumbers title="greeting_guestbook/main.py"
-import logging
+@DBOS.step()
+def step_one():
+    print("Step one completed!")
 
-import requests
-from dbos import get_dbos_database_url
-from fastapi import FastAPI
-from sqlalchemy import create_engine
+@DBOS.step()
+def step_two():
+    print("Step two completed!")
 
-from .schema import dbos_hello
+@DBOS.workflow()
+def dbos_workflow():
+    step_one()
+    step_two()
 
-app = FastAPI()
-logging.basicConfig(level=logging.INFO)
+if __name__ == "__main__":
+    DBOS.launch()
+    dbos_workflow()
+```
+
+In DBOS, you write programs as **workflows** of **steps**.
+Workflows and steps are ordinary Python functions annotated with the `@DBOS.workflow()` and `@DBOS.step()` decorators.
+DBOS **durably executes** workflows, persisting their state to a database so if they are interrupted or crash, they automatically recover from the last completed step.
 
-# Sign the guestbook using an HTTP POST request
-def sign_guestbook(name: str):
-    requests.post(
-        "https://demo-guestbook.cloud.dbos.dev/record_greeting",
-        headers={"Content-Type": "application/json"},
-        json={"name": name},
-    )
-    logging.info(f">>> STEP 1: Signed the guestbook for {name}")
+Run this code with `python3 main.py` and it should print output like:
 
-# Create a SQLAlchemy engine.
-engine = create_engine(get_dbos_database_url())
+```shell
+13:47:09 [    INFO] (dbos:_dbos.py:272) Initializing DBOS
+13:47:09 [    INFO] (dbos:_dbos.py:401) DBOS launched
+Step one completed!
+Step two completed!
+```
 
-# Record the greeting in the database using SQLAlchemy
-def insert_greeting(name: str) -> str:
-    with engine.begin() as sql_session:
-        query = dbos_hello.insert().values(name=name)
-        sql_session.execute(query)
-    logging.info(f">>> STEP 2: Greeting to {name} recorded in the database!")
+To see durable execution in action, let's modify the app to serve a DBOS workflow from an HTTP endpoint using FastAPI.
```

Comment:
I understand that you put FastAPI here to demonstrate recovery.
But maybe it's better to talk about FastAPI integration in a separate section. Something like "FastAPI Integration", and also explains briefly about the lifecycle hook.